File: Java/InsertionSort.java
import java.util.Arrays;

public class InsertionSort {
    public static void main(String[] args) {
        int[] arr = {2, 55, 7, 4, 13, 17};
        // Calling the method
        InsertionSort(arr);
        System.out.println(Arrays.toString(arr));
    }

    // Method for Insertion Sort
    static void InsertionSort(int[] num){
        for (int i = 0; i < num.length-1; i++) {
            for (int j = i+1; j >0 ; j--) {
                if (num[j] < num[j-1]){
                    int temp = num[j];
                    num[j] = num[j-1];
                    num[j-1] = temp;
                }
                else {
                    break;
                }
            }
        }
    }
}

File: C++/Insertion Sort.cpp
#include<iostream>
using namespace std;

int InsertionSort(int arr[], int n){

   for(int i =1; i<n; i++){
    int temp = arr[i];
    int j =i-1;
     for(; j>=0; j--){
        if(arr[j]>temp){
            arr[j+1] = arr[j];
        }
        else{
            break;
        }
    }
    arr[j+1] = temp;
    }
    cout<< "Insertion Sorted Array : " << endl;
    for(int i = 0; i<n; i++ ){
        cout<< arr[i] << " ";
    }
}
int main(){
    int arr[5] = {6,5,4,2,3};
    cout << InsertionSort(arr, 5);

}
